clang/test/Analysis/store-to-immutable-basic.cpp

```diff
@@ -0,0 +1,166 @@
+// RUN: %clang_analyze_cc1 -analyzer-checker=alpha.core.StoreToImmutable -verify %s
+
+// Test basic functionality of StoreToImmutable checker
+// This tests direct writes to immutable regions without function modeling
+
+// Direct write to a const global variable
+const int global_const = 42; // expected-note {{Memory region is in immutable space}}
```

Comment:
All of the global variables mentioned has an explicit initialized.
Could you add tests that only declares the global (aka. tentative definition).
```c++
const int tentative_global_const; // no explicit initializer, nor `extern` keyword.
```

Reply:
This would not even compile in C++, so I have created 2 test files and split the test cases.
I have put those that can be expressed in C without compilation error in one test file, and put the C++ specific parts in another.